pkg/core/dao/cacheddao.go

```diff
@@ -148,6 +156,9 @@ func (cd *Cached) Persist() (int, error) {
 	if ok {
 		var simpleCache *Simple
 		for simpleCache == nil {
+			if err := lowerCache.FlushStorage(); err != nil {
+				return 0, err
+			}
```

Comment:
Why do you need to flush `lowerCache`?

Reply:
Because `FlushStorage` only flushes current `storage` calling `Put`/`Delete` of the underlying `DAO`. That DAO will also cache results if it is `CachedDAO`. But in `Persist` we need items to go all the way down to `MemCached`.

Reply:
> Because FlushStorage only flushes current storage calling Put/Delete of the underlying DAO

But isn't that what we want for `Cached` over `Cached`?

Reply:
This depends on what we expect `Persist()` to do. I'd expect it to flush _everything_ to a persistent store. In our case we have only 2 levels of `*Cached` and lower of them is persisted separately so it should work too.

Reply:
It's expected to flush things exactly one layer below, so if you have D1->CD1->CD2 hierarchy (which is typical) and call `CD2.Persist()` it should flush changes from CD2 to CD1 (resulting in `CD1 == CD2` semantically), but shouldn't affect D1 to CD1 relations in any way.

Reply:
So I still think it's not correct.

Reply:
Fixed.

Reply:
So it's fixed and it's still there, how come?

Reply:
I guess it only works if you don't have more than 2 layers of Cached, but that's the case we have. And it flushes storage items to the (memcached) DB, but that's probably what we need for proper `Find` ordering (I've tried removing this flush and everything failed quickly).